runtime/src/iree/hal/local/executable_plugin.h

```diff
@@ -0,0 +1,440 @@
+
+// Defines a set of supported sanitizers that plugins may be compiled with.
+// The runtime uses this declaration to check as to whether the plugin is
+// compatible with the hosting environment for cases where the sanitizer
+// requires host support.
+typedef enum iree_hal_executable_plugin_sanitizer_kind_e {
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_NONE = 0,
+  // Indicates the plugin is compiled to use AddressSanitizer:
+  // https://clang.llvm.org/docs/AddressSanitizer.html
+  // Equivalent compiler flag: -fsanitize=address
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_ADDRESS = 1,
+  // Indicates the plugin is compiled to use MemorySanitizer:
+  // https://clang.llvm.org/docs/MemorySanitizer.html
+  // Equivalent compiler flag: -fsanitize=memory
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_MEMORY = 2,
+  // Indicates the plugin is compiled to use ThreadSanitizer:
+  // https://clang.llvm.org/docs/ThreadSanitizer.html
+  // Equivalent compiler flag: -fsanitize=thread
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_THREAD = 3,
+  // Indicates the plugin is compiled to use UndefinedBehaviorSanitizer:
+  // https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+  // Equivalent compiler flag: -fsanitize=undefined
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_UNDEFINED = 4,
+
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_MAX_ENUM = INT32_MAX,
+} iree_hal_executable_plugin_sanitizer_kind_t;
+
+// The sanitizer kind currently enabled in the compilation unit. Can be used
+// in plugins to initialize the sanitizer header field.
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND \
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_ADDRESS
+#endif  // __has_feature(address_sanitizer)
+#if __has_feature(memory_sanitizer)
+#define IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND \
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_MEMORY
+#endif  // __has_feature(memory_sanitizer)
+#if __has_feature(thread_sanitizer)
+#define IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND \
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_THREAD
+#endif  // __has_feature(thread_sanitizer)
+#endif  // defined(__has_feature)
+#if !defined(IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND)
+#define IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND \
+  IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_NONE
+#endif  // !IREE_HAL_EXECUTABLE_PLUGIN_SANITIZER_KIND
+
+//===----------------------------------------------------------------------===//
+// Versioning and interface querying
+//===----------------------------------------------------------------------===//
+
+// Version code indicating the minimum required runtime structures.
+// Runtimes cannot load plugins with newer versions but may be able to load
+// older versions if backward compatibility is enabled.
+//
+// NOTE: until we hit v1 the versioning scheme here is not set in stone.
+// We may want to make this major release number, date codes (0x20220307),
+// or some semantic versioning we track in whatever spec we end up having.
```

Comment:
Did you intend for this example date code to be the date you wrote the code? 2023 not 2022 in that case :P

```suggestion
// NOTE: until we hit v1 the versioning scheme here is not set in stone.
// We may want to make this major release number, date codes (0x20230307),
// or some semantic versioning we track in whatever spec we end up having.
```

Reply:
it's just an example :)
![image](https://user-images.githubusercontent.com/75337/231875772-1dda3267-7069-4155-8cd4-6bddddbbe444.png)